code/src/components/CategoryMap.js

```diff
@@ -0,0 +1,184 @@
+import React from "react"
+import styled from 'styled-components'
+import { useSelector, useDispatch } from 'react-redux'
+import { todos } from '../reducers/todos'
+import { ToDoItem } from './ToDoItem'
+import { EmptyDisplay } from './EmptyDisplay'
+import { TextButton } from '../lib/TextButton'
+import { FlexColumn } from '../lib/FlexColumn'
+import Bounce from 'react-reveal/Bounce'
+import LightSpeed from 'react-reveal/LightSpeed'
+import img from '../Assets/dropdown.svg'
+
+const FolderContainer = styled.div`
+  width: 100%;
+  border-bottom: 3px solid #f0f0f0;
+  padding: 16px;  
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  background-color: ${(props) => props.backgroundColor};
+  font-size: 20px;
+  color: #a3a3a3;
+  transition: all 1s ease-in-out;
+  &:hover {
+    background-color: papayawhip;
+  }
+`
+
+const LeftSideContainer = styled.div`
+  display: flex;
+  align-items: center;
+`
+
+const FolderButton = styled.button`
+  font-size: ${(props) => props.fontSize};
+  color: #a3a3a3;
+  display: flex;
+  align-items: center;
+  background: transparent;
+  border: none;
+  cursor: pointer;
+  text-align: left;
+  @media (min-width: 1024px) {
+    ont-size: 20px;
+  }
+`
+
+const ButtonText = styled.span`
+  width: 100px;
+  @media (min-width: 660px) {
+    width: 250px;
+  }
+`
+
+const Image = styled.img`
+  height: 20px;
+  width: 20px;
+  margin-right: 8px;
+`
+
+const Dot = styled.div`
+  background-color: red;
+  border-radius: 100%;
+  height: 10px;
+  width: 10px;
+`
+const Dropdown = styled.select`
+  background: transparent;
+  margin: 0;
+  line-height: 1;
+  border: 0;
+  padding: 0;
+  border-radius: 0;
+  width: 130%;
+  z-index: 10;
+  font-size: 13px;
+  color: black;
+`
+const DropdownContainer = styled.div`
+  display: block;
+  font-size: 13px;
+  padding: 8px 14px 8px 5px;
+  background: transparent;
+  border: 1px solid #ccc;
+  border-radius: 6px;
+  overflow-x: hidden;
+  background-image: url(${img});
+  background-repeat: no-repeat;
+  background-size: 15px;
+  background-position: 95% center; 
+  margin-left: 8px;
+  @media (min-width: 660px) {
+     background-size: 20px;
+   }
+`
+export const CategoryMap = () => {
+    const dispatch = useDispatch()
+    const folders = useSelector((store) => store.todos.list.categories)
+
+    const openFolder = (itemIndex, item) => {
+        dispatch(
+            todos.actions.revealItem({ itemIndex: itemIndex, reveal: !item.reveal })
```

Comment:
when key and value are the same `itemIndex: itemIndex`, you can shorten it to only write `itemIndex` once